pkg/util/util.go

```diff
@@ -312,6 +317,75 @@ func getAPIUrlsForOpenShiftV4() (apiUrl string, apiInternalUrl string, err error
 	return apiUrl, apiInternalUrl, nil
 }
 
+func GetRouterCanonicalHostname(client client.Client, namespace string) (string, error) {
```

Comment:
Why do we need this?

Reply:
To support database patching in case of different restoring namespace within the same cluster.